### PR TITLE
Typo in testing script

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

**Review by douzzer — CHANGES_REQUESTED** [review]
Thanks for noticing this!

Now we know that sub-sub-test is broken -- obviously need to fix ocspstapling1 before merging the script fix.  Btw there's also this in the picture, for context:
```
    # must disable ocspstapling1 and featuresets that depend on it -- ASAN/leak bugs.
    SANITIZER_CONFIG_FLAGS_COMMON=(
	--disable-ocspstapling
	--disable-nginx
	--disable-wpas
	--disable-haproxy
	--disable-lighty
	--disable-curl
    )
```

Re my bad operator in the script: we hereby discover that `shellcheck --severity=error` is blind to this, which seems wrong.  Default `shellcheck` sees it:
```
In scripts/ocsp-stapling.test line 238:
    if [ $? -neq 0 ]; then
         ^-- SC2181 (style): Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
            ^--^ SC2057 (warning): Unknown binary operator.
```
Unfortunately default `shellcheck` is otherwise unnecessarily noisy.  But I'll come up with some combination of frobs to the `shellcheck` options and the scripts to let us detect more defects, including this one.

**Comment by julek-wolfssl** [comment]
Thanks for the reviews @kaleb-himes @douzzer . I didn't realize this would uncover more issues. I'll get back to you after I look into the failures.

**Comment by kaleb-himes** [comment]
After looking a little closer at the test scenario the zero is a failure case (it is expected to fail) but when --enable-all is configure that case is succeeding (WOLFSSL_SUCCESS or 1).

Correction on report:
`./configure --enable-ocspstapling && make && ./scripts/ocsp-stapling.test` -> correctly fails
`./configure --enable-all && make && ./scripts/ocsp-stapling.test` -> incorrectly succeeds (according to the test case...)

Suggestion:
Is the test case bad IE should that ALWAYS fail or only sometimes fail? The test case might have incorrect assumptions depending on how the library was configured!

**Review by kaleb-himes — APPROVED** [review]
Very much like this solution! Accounts for the case when s_client aborts before the verify can take place (and therefore fail in the expected way).

**Comment by kaleb-himes** [comment]
Great fix @julek-wolfssl (and great initial find!)

File: scripts/ocsp-stapling.test [hunk 1]
```diff
@@ -234,8 +234,12 @@ if [ ! -f "$ready_file" ]; then
...
 else
     # should fail if ocspstapling is also enabled
-    echo "hi" | openssl s_client -status $V4V6_FLAG -connect ${LOCALHOST}:$port1 -cert ./certs/client-cert.pem -key ./certs/client-key.pem -CAfile ./certs/ocsp/root-ca-cert.pem 2>&1 | tee /dev/stderr | fgrep -q 'self signed certificate in certificate chain'
-    if [ $? -neq 0 ]; then
+    OPENSSL_OUTPUT=$(echo "hi" | openssl s_client -status $V4V6_FLAG -connect ${LOCALHOST}:$port1 -cert ./certs/client-cert.pem -key ./certs/client-key.pem -CAfile ./certs/ocsp/root-ca-cert.pem 2>&1)
+    OPENSSL_RESULT=$?
+    echo "$OPENSSL_OUTPUT"
+    fgrep -q 'self signed certificate in certificate chain' <<< "$OPENSSL_OUTPUT"
+    FGREP_RESULT=$?
+    if [ $OPENSSL_RESULT -eq 0 -a FGREP_RESULT -ne 0 ]; then
```
Comment (douzzer) [inline scripts/ocsp-stapling.test:242]: whoops!  you did `FGREP_RESULT -ne 0` in the test, instead of `$FGREP_RESULT -ne 0`.

**Review by douzzer — CHANGES_REQUESTED** [review]
also might want to add some double-quoting around the subcommand in line 237, for safety.

File: scripts/ocsp-stapling.test (hunk 1 above)
Comment (julek-wolfssl) [reply scripts/ocsp-stapling.test]: Very good catch. Fixed and pushed.

**Comment by julek-wolfssl** [comment]
@douzzer I added quotes around the variable parameters on line 237. Is that what you had in mind?

**Comment by douzzer** [comment]
> @douzzer I added quotes around the variable parameters on line 237. Is that what you had in mind?

That's good too, but no I meant double-quoting the entire subcommand, à la
```
foo="$(echo bar)"
```
however, `shellcheck` doesn't seem to think there's anything wrong with
```
foo=$(ls)
```
and experimentally, embedded whitespace in the subcommand output is captured without incident, so what you have is good (ignore my pedantry:-).

**Review by douzzer — APPROVED** [review]
LGTM
```
$ wolfssl-multi-test.sh [...] quick-check
[...]
final tally: all 22 selected checks succeeded for b4d17eb82c.
```

**Pull request merged**